baremaps-geoparquet/src/main/java/org/apache/baremaps/geoparquet/GeoParquetReader.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.baremaps.geoparquet;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.apache.baremaps.geoparquet.data.*;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.avro.AvroReadSupport;
+import org.apache.parquet.column.page.PageReadStore;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.FileMetaData;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+import org.apache.parquet.io.ColumnIOFactory;
+import org.apache.parquet.io.RecordReader;
+import org.apache.parquet.schema.MessageType;
+
+
+/**
+ * This reader is based on the parquet example code located at: org.apache.parquet.example.data.*.
+ */
+public class GeoParquetReader {
+
+  private final URI uri;
+
+  private Configuration configuration;
+
+  private Map<FileStatus, FileInfo> metadata = new LinkedHashMap<>();
+
+  public GeoParquetReader(URI uri) {
+    this.uri = uri;
+    this.initialize();
+  }
+
+  public void initialize() {
+    try {
+      this.configuration = getConfiguration();
+
+      // List all the files that match the glob pattern
+      Path globPath = new Path(uri.getPath());
+      URI rootUri = getRootUri(uri);
+      FileSystem fileSystem = FileSystem.get(rootUri, configuration);
+      List<FileStatus> files = Arrays.asList(fileSystem.globStatus(globPath));
+
+      // Read the metadata of each file
+      for (FileStatus fileStatus : files) {
+
+        // Open the Parquet file
+        try (ParquetFileReader reader = ParquetFileReader
```

Comment:
Why not use the `ParquetReader`? It looks like `ParquetFileReader` is an internal implementation: `Internal implementation of the Parquet file reader as a block container`. https://www.javadoc.io/doc/org.apache.parquet/parquet-hadoop/latest/org/apache/parquet/hadoop/ParquetFileReader.html

`ParquetReader` goal is to read records from a Parquet file.
https://www.javadoc.io/doc/org.apache.parquet/parquet-hadoop/latest/org/apache/parquet/hadoop/ParquetReader.html